test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -685,6 +690,26 @@
   "Support fact command versions"
   [:v4])
 
+(defn approx-facts-query []
+  "Get all fact names and their values converting values to text.
+   Returns a vector of maps following the format below:
+   [{:certname \"blarg\" :name \"fact1\" :value \"1\"}]"
+  (->> [" SELECT fp.path as name,"
+        "        COALESCE(fv.value_string,"
+        "                 cast(fv.value_integer as text),"
+        "                 cast(fv.value_boolean as text),"
+        "                 cast(fv.value_float as text),"
+        "                 '') as value,"
+        "        fs.certname"
+        " FROM factsets fs"
+        "      INNER JOIN facts as f on fs.id = f.factset_id"
+        "      INNER JOIN fact_values as fv on f.fact_value_id = fv.id"
+        "      INNER JOIN fact_paths as fp on f.fact_path_id = fp.id"
+        " WHERE fp.depth = 0"
+        " ORDER BY name ASC"]
+       (apply str)
```

Comment:
Hmm, wondering about changing this to "(apply str/join "\n")" so that we don't need the leading spaces, and assuming it works, that might also make the sql log output nicer (though not postiive about that).  Or we can just join on " ".